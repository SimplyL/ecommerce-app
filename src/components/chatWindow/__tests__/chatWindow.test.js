import React from 'react';
import { shallow } from 'enzyme';

import ChatWindow from '../index';

describe('ChatWindow component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ChatWindow />);
  });

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should open chat window on button click', () => {
    wrapper.find('ChatButton').simulate('click');
    expect(wrapper.state().isOpen).toBe(true);
    expect(wrapper.find('ChatContainer')).toBeDefined();
  });

  it('should open chat and add message', () => {
    wrapper.find('ChatButton').simulate('click');
    expect(wrapper.state().messages.length).toBe(1);
    wrapper.setState({ inputValue: 'test' });
    wrapper.find('TextInput').simulate('keyPress', { key: 'Enter' });
    expect(wrapper.state().messages.length).toBe(3);
  });
});
