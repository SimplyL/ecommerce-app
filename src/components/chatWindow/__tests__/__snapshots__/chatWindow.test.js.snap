// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatWindow component should render without crashing 1`] = `
<Container>
  <ChatButton
    onClick={[Function]}
  >
    +
  </ChatButton>
</Container>
`;
